doc/use_cases/turtlebot3_tutorial.md

```diff
@@ -0,0 +1,239 @@
+
+If Gazebo fails to start, run the following commands and try to launch Gazebo again.
+
+    killall gzserver
+    killall gzclient
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/gazebo_turlebot1.PNG?raw=true)
+
+### 2- Launch Turtlebot 3 Robot State Publisher
+ 
+Launch Turtlebot 3 specific nodes,
+
+    ros2 launch turtlebot3_bringup turtlebot3_state_publisher.launch.py use_sim_time:=True
+
+Make sure use_sim_time is set to **True**.
+
+### 3- Launch Navigation2 
+
+Launch Navigation 2. If you set `autostart:=False`, you need to click on the start button in RVIZ to initialize the nodes. Make sure `use_sim time` is set to **True**, because we want to use the time simulation time in Gazebo instead of the system time.
+
+turtlebot3_world.yaml is the configuration file for the map we want to provide Navigation2. In this case, it has the map resolution value, threshold values for obstacles and free spaces, and a map file location.
+
+```
+ros2 launch nav2_bringup nav2_bringup_launch.py use_sim_time:=True autostart:=False \
+    map:=/home/<user_name>/navigation2_ws/install/nav2_bringup/share/nav2_bringup/launch/turtlebot3_world.yaml
+```
+Again don't forget to change **<user_name>** to your actual user name. 
+
+### 4-  Launch RVIZ
+
+Launch RVIZ with a pre-defined configuration file.
+
+    ros2 run rviz2 rviz2 -d $(ros2 pkg prefix nav2_bringup)/share/nav2_bringup/launch/nav2_default_view.rviz
+
+Now, you should see a shadow of Turtlebot 3 robot model in the center of the plot in Rviz. Click on the Start button (Bottom Left) if you set the auto_start parameter to false.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/Rviz_start.PNG?raw=true)
+
+Now, the map should appear in Rviz.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/Rviz_initial.PNG?raw=true)
+
+### 5- Initialize the location of Turtlebot 3
+
+First, find where the robot is in Gazebo. You can see where the robot's initial position in Gazebo.
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/gazebo_turlebot1.PNG?raw=true)
+
+Set the pose of the robot in Rviz. Click on the 2D Pose Estimate button and point the location of the robot on the map. The direction of the green arrow is the orientation of Turtlebot.
+  
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/Rviz2D_Pose_Estimate.PNG?raw=true)
+
+Now, the 3D model of Turtlebot should move to that location. A small error in the estimated location is tolerable.
+
+### 6-  Send a Goal Pose
+
+Pick a target location for Turtlebot on the map. You can send Turtlebot 3 a goal position and a goal orientation by using the **Navigation2 Goal** the **GoalTool** buttons.
+
+*Note*: Difference between buttons. Navigation2 Goal button uses a ROS2 Action to send the goal and GoalTool publishes the goal to a topic.
```

Comment:
Good note, but needs a "why use 1 vs other" which even I don't have a good answer for right now